src/ui/components/Comments/TranscriptComments/types.ts

```diff
@@ -0,0 +1,7 @@
+import { Comment, Reply } from "ui/state/comments";
+
+export type CommentData =
+  | { type: "comment"; comment: Comment }
+  | { type: "new_comment"; comment: Comment }
+  | { type: "reply"; comment: Reply }
+  | { type: "new_reply"; comment: Reply };
```

Comment:
CommentData is now a top level type because it needs to be shared with components that use it